queries/flights.py

```diff
@@ -0,0 +1,502 @@
+"""
+
+    Greynir: Natural language processing for Icelandic
+
+    Flight schedule query response module
+
+    Copyright (C) 2021 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program. If not, see http://www.gnu.org/licenses/.
+
+
+    This module handles queries relating to air travel.
+
+"""
+
+import re
+import random
+import logging
+import cachetools
+from html import escape
+from typing import List, Dict, Any, Optional
+from datetime import datetime, timedelta, timezone
+
+from query import Query, QueryStateDict
+from queries import query_json_api
+from tree import Result
+from settings import changedlocale
+
+from reynir import NounPhrase
+from geo import capitalize_placename, iceprep_for_placename, icelandic_city_name
+
+
+_FLIGHTS_QTYPE = "Flights"
+
+
+TOPIC_LEMMAS = [
+    "flugvél",
+    "flugvöllur",
+    "flug",
+    "lenda",
+    "brottfarartími",
+    "lendingartími",
+]
+
+
+def help_text(lemma: str) -> str:
+    """Help text to return when query.py is unable to parse a query but
+    one of the above lemmas is found in it"""
+    return "Ég get svarað ef þú spyrð til dæmis: {0}?".format(
+        random.choice(
+            (
+                "Hvenær lendir næsta vél frá Kaupmannahöfn",
+                "Hvenær fer næsta flug til Lundúna",
+            )
+        )
+    )
+
+
+# This module wants to handle parse trees for queries
+HANDLE_TREE = True
+
+# The grammar nonterminals this module wants to handle
+QUERY_NONTERMINALS = {"QFlights"}
+
+# The context-free grammar for the queries recognized by this plug-in module
+GRAMMAR = """
+
```

Comment:
Would be good to have comments here with examples of the supported query forms.

Reply:
Good point! I added a few examples to the grammar, do you think more would be necessary/helpful?